### PR TITLE
fix: add run first determines whether a package.json exists (#138)

### PR DESCRIPTION
**Pull request by wChenonly**
Prompt the user if it is not currently a valid node project 

image

Linked Issues
#138

### CODE REVIEW

File: src/fs.ts [hunk 1]
```diff
@@ -17,8 +17,24 @@ export function getPackageJSON(ctx?: RunnerContext): any {
...
   }
 }
+
+function getPath(ctx?: RunnerContext): string {
+  const cwd = ctx?.cwd ?? process.cwd()
+  const path = resolve(cwd, 'package.json')
+  return path
+}
+
+export function isExistsPackageJSON(): boolean {
```
Comment (FrankFang) [inline src/fs.ts:31]: Maybe `packageJsonExists` is better.

**Comment by antfu** [comment]
I am not really convinced by this. I'd expect `ni` to work in all cases, when package.json is not presented it should show a prompt to choose the agent. I'd love to see a minimal reproduction of the original issue as I didn't saw that error.

**Comment by wChenonly** [comment]
> I am not really convinced by this. I'd expect `ni` to work in all cases, when package.json is not presented it should show a prompt to choose the agent. I'd love to see a minimal reproduction of the original issue as I didn't saw that error.

`nr`is ok,`ni`is wrong, you can create an empty folder at random, go to the end point and execute`ni`.  Including `nu` is also unreasonable
Jietu20230616-205530-HD

**Comment by antfu** [comment]
https://github.com/antfu/ni/issues/138#issuecomment-1594894106

**Pull request closed without merging**